### PR TITLE
Added Josephus Queries problem

### LINKED ISSUE #207: [Feature Request] Josephus Queries - Recently Asked Interview Question at Apple
### Is your feature request related to a problem? Please describe.
The problem relates to a scenario, where people are arranged in a circle, and every k-th person is eliminated until only one remains. The goal is to solve variations of the Josephus problem efficiently using modern techniques, making it relevant to coding interviews at top companies like Apple.

### Describe the solution you'd like
I propose solving this problem based on: 

- The position of the last person standing for any circle size and step value.
- Intermediate positions after a certain number of eliminations.
- Variants like when the counting direction alternates or changes after certain eliminations.

The solution involves recursive solution.

Would solve the problem in C++, Java, Python, Javascript, Go and clearly explain each and every step and add key points to mention at the interview as well!

---

**Would you like to work on this feature?**  
- [x] Yes  
- [ ] No

### PR DESCRIPTION
**Pull request by J-B-Mugundh**
### Description
Added Josephus Queries solution. Elaborated the solution of recursion in C++, Java, Python, Javascript, Go

Fixes #207 

### Type of change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by J-B-Mugundh** [comment]
@ajay-dhangar I'm not sure why build is failing! Could you please check!

**Comment by ajay-dhangar** [comment]
> @ajay-dhangar I'm not sure why build is failing! Could you please check!

@J-B-Mugundh, please sync your fork with the latest changes from the main repository. I've resolved the `backtracking-questions.md` error at line 56.

Here's what you need to do:
1. Fetch the latest updates from the main repository:
   ```
   git fetch upstream
   git merge upstream/main
   ```
2. Push the merged changes to your fork:
   ```
   git push origin your-branch
   ```
3. After syncing, resolve any remaining issues.

Please don't hesitate to let me know if you encounter any problems.

**Comment by J-B-Mugundh** [comment]
@ajay-dhangar Updated bro!

File: docs/Recursion/Josephus-Queries.md [hunk 1]
```diff
@@ -0,0 +1,172 @@
...
+Output:
+2
+6
+1
+1107
+```
+
+## Constraints
+- \(1 \leq q \leq 100,000\)
+- \(1 \leq k \leq n \leq 1,000,000,000\)
```
Comment (ajay-dhangar) [inline docs/Recursion/Josephus-Queries.md]: replace 

```md
-  \(1 \leq q \leq 100,000\)
- \(1 \leq k \leq n \leq 1,000,000,000\)
```

with

```md
- $1 \leq q \leq 100,000$
- $1 \leq k \leq n \leq 1,000,000,000$
```

for best practice

File: docs/Recursion/Josephus-Queries.md [hunk 2]
```diff
@@ -0,0 +1,172 @@
...
+
+## Solution
+This solution employs a recursive approach to solve the Josephus Queries problem.
+
+### Recursive Approach
+The problem can be defined recursively as follows:
+
+1. If there is only one child, the only child present is the one that is removed.
+2. For more than one child, the \(k\)-th removal can be computed by reducing the problem size by 1 each time we remove a child.
```
Comment (ajay-dhangar) [inline docs/Recursion/Josephus-Queries.md]: replace `\(k\)-th` with `$k^{th}$`

File: docs/Recursion/Josephus-Queries.md [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+### Recursive Function Definition
+Let `josephus(n, k)` be the function that returns the position of the \(k\)-th child to be removed when there are \(n\) children.
```
Comment (ajay-dhangar) [inline docs/Recursion/Josephus-Queries.md]: replace `\(k\)-th` with `$k^{th}$` and `\(n\)` with `$n$`

File: docs/Recursion/Josephus-Queries.md [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+
+#### Base Case
+- If \(n = 1\), return \(1\) (the only child).
```
Comment (ajay-dhangar) [inline docs/Recursion/Josephus-Queries.md]: replace `\(n = 1\)` with `$n = 1$` and `\(1\)` with `$1$`

File: docs/Recursion/Josephus-Queries.md [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+
+#### Recursive Case
+- If there are more than one child, the position of the \(k\)-th child can be found using:
+  \[
+  josephus(n, k) = (josephus(n-1, k) + k) \% n
+  \]
```
Comment (ajay-dhangar) [inline docs/Recursion/Josephus-Queries.md]: replace 

```md
\[
  josephus(n, k) = (josephus(n-1, k) + k) \% n
 \]
```
with

```md
$$
josephus(n, k) =
\begin{cases} 
0 & \text{if } n = 1, \\
(josephus(n-1, k) + k) \% n & \text{if } n > 1.
\end{cases}
$$
```

File: docs/Recursion/Josephus-Queries.md [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+  This gives the position in the new configuration after one child has been removed.
+
+### Final Adjustment
+Since our function will return a 0-based index, we will need to add \(1\) to convert it back to a 1-based index.
+
+## Time and Space Complexity
+- **Time Complexity:** \(O(n)\), where \(n\) is the number of children.
+- **Space Complexity:** \(O(n)\) due to the recursive stack space.
+
```
Comment (ajay-dhangar) [inline docs/Recursion/Josephus-Queries.md]: replace `\(O(n)\)` with `$O(n)$` and `\(n\)` with `$n$`

File: docs/Recursion/Josephus-Queries.md [hunk 7]
```diff
@@ -0,0 +1,172 @@
...
+		var n, k int
+		fmt.Scan(&n, &k)
+		fmt.Println(josephus(n, k))
+	}
+}
+```
+
+## Conclusion
+The Josephus problem presents a fascinating exploration of recursive algorithms and their efficiency in solving complex queries. By employing a recursive approach, we can derive the position of the \(k\)-th child to be removed in a circle of \(n\) children efficiently. Despite the recursive nature of the solution leading to a time complexity of \(O(n)\) and a space complexity of \(O(n)\), it effectively highlights the power of recursion in algorithm design.
+
```
Comment (ajay-dhangar) [inline docs/Recursion/Josephus-Queries.md]: also, update the LaTeX-style math formula

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions for good practices

**Comment by J-B-Mugundh** [comment]
@ajay-dhangar Updated bro! Thanks so for much for your time and efforts!

**Review by ajay-dhangar — APPROVED** [review]
Done

**Pull request merged**